tidy3d/components/geometry/triangulation.py

```diff
@@ -0,0 +1,134 @@
+from typing import List, Tuple
+
+import numpy as np
+import shapely
+
+from ..types import ArrayFloat1D, ArrayFloat2D
+
+
+class Vertex:
```

Comment:
why not make this a Tidy3DBaseModel? just for consistency and documentation?

or even just a dataclass if that's easier.

Reply:
I guess this is supposed to be internal-use only. We should be able to replace it with something more efficient as soon as we find something without worrying about backwards compatibility of the interface. I've replaced with dataclass just in case.